internal/docker/docker.go

```diff
@@ -200,3 +207,23 @@ func (c *Docker) runCommand(cmdName string, args ...string) error {
 	slog.Info(fmt.Sprintf("=== Docker end %s", strings.Repeat("=", 65)))
 	return err
 }
+
+func toGenerateRequestJSON(state *config.PipelineState, filepath string) error {
+	data, err := json.MarshalIndent(state, "", "  ")
+	if err != nil {
+		return errors.New("failed to marshal state to JSON")
```

Comment:
It is almost always a good idea to preserve the original error context when returning errors in Go. The language has an error wrapping mechanism. For example this error would turn into:
```suggestion
		return fmt.Errorf("failed to marshal state to JSON: %w", err)
```

Reply:
Done.